forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -9,20 +9,70 @@ use anyhow::{bail, Context, Result};
 use forc_pkg::{self as pkg, PackageManifestFile};
 use fuel_core_client::client::types::TransactionStatus;
 use fuel_core_client::client::FuelClient;
-use fuel_tx::{Output, TransactionBuilder};
+use fuel_tx::{Output, Salt, TransactionBuilder};
 use fuel_vm::prelude::*;
 use futures::FutureExt;
 use pkg::BuiltPackage;
-use std::path::PathBuf;
 use std::time::Duration;
+use std::{collections::BTreeMap, path::PathBuf};
 use sway_core::language::parsed::TreeType;
 use sway_core::BuildTarget;
 use tracing::info;
 
+#[derive(Debug)]
 pub struct DeployedContract {
     pub id: fuel_tx::ContractId,
 }
 
+type ContractSaltMap = BTreeMap<String, Salt>;
+
+/// Takes the salt input passed via the --salt option, validates them against
+/// the manifests and returns a ContractSaltMap (BTreeMap of contract names to salts).
+fn validate_and_parse_salts<'a>(
+    salt_args: Vec<String>,
+    manifests: impl Iterator<Item = &'a PackageManifestFile>,
+) -> Result<ContractSaltMap> {
+    let mut contract_salt_map = BTreeMap::default();
+
+    // Parse all the salt arguments first, and exit if there are errors in this step.
+    for salt_arg in salt_args {
+        if let Some((given_contract_name, salt)) = salt_arg.split_once(':') {
+            if contract_salt_map.contains_key(given_contract_name) {
+                bail!("2 salts provided for contract '{given_contract_name}'");
+            }
+            contract_salt_map.insert(given_contract_name.to_string(), salt_from_str(salt)?);
+        } else {
+            bail!("Invalid salt provided - salt must be in the form <CONTRACT_NAME>:<SALT> when deploying a workspace");
+        }
+    }
+
+    for manifest in manifests {
+        for dep in manifest.contract_deps() {
+            if let Some(declared_salt) = contract_salt_map.get(dep.0) {
+                bail!(
+                    "Redeclaration of salt using the option '--salt' while a salt exists for contract '{}' \
+                    under the contract dependencies of the Forc.toml manifest for '{}'\n\
+                    Existing salt: '0x{}',\nYou declared: '0x{}'\n",
+                    dep.0,
+                    manifest.project_name(),
+                    dep.1.salt,
+                    declared_salt,
+                    );
+            }
+        }
+    }
+
+    Ok(contract_salt_map)
+}
+
+/// Decodes a hexadecimal number into a fuel_tx::Salt.
+fn salt_from_str(salt_str: &str) -> Result<Salt> {
+    let mut bytes = [0u8; 32];
+    hex::decode_to_slice(salt_str.replace("0x", ""), &mut bytes)?;
+
+    Ok(Salt::new(bytes))
+}
```

Comment:
I _think_ `Salt` impls `FromStr`, so we should be able to remove this in favour of using:
```rust
let salt: Salt = salt_str.parse()?;
```